### ISSUE TITLE
morphology.binary_closing() is ~50x slower than morphology.closing(). The ratio should be reversed

### ISSUE THREAD

**Issue by vali19th** [scikit-image/scikit-image#7458]
### Description:

I tried with both of these images:
[0_and_255.png](https://github.com/scikit-image/scikit-image/assets/77642711/3307fce9-e493-4e8d-a3b3-0970c713363e)
[0_and_1.png](https://github.com/scikit-image/scikit-image/assets/77642711/6a5d90e0-2453-41a1-a580-af9297d146fb)

Docs saying it should be faster: https://scikit-image.org/docs/stable/api/skimage.morphology.html#skimage.morphology.binary_closing

### Way to reproduce:

```python
import time
import numpy as np
from skimage import io, morphology

img = io.imread("tmp/x.png")[:, :, 0]
print(np.unique(img), img.shape, img.dtype)  # [0 1] (3508, 2480) uint8
kernel = np.ones((35, 35))

t = time.perf_counter()
morphology.binary_closing(img, kernel)
t2 = time.perf_counter()
morphology.closing(img, kernel)
t3 = time.perf_counter()

print(t2 - t)  # 5.420551682997029
print(t3 - t2)  # 0.08474353399651591
```

### Version information:

```Shell
3.11.9 (main, Apr 19 2024, 16:48:06) [GCC 11.2.0]  # conda environment
Linux-6.5.0-41-generic-x86_64-with-glibc2.35
scikit-image version: 0.23.2
numpy version: 1.26.4
```

**Comment by michaelbratsch** [comment]
The same is true for `morphology.opening` and `morphology.binary_opening`.

It is an issue with the underlying scipy implementation see https://github.com/scipy/scipy/issues/13991

There are also similar discussion in https://github.com/scikit-image/scikit-image/issues/1190, for example: https://github.com/scikit-image/scikit-image/issues/1190#issuecomment-572384985

Maybe it makes sense to change the documentation and not confuse people that the binary operations are faster.

**Comment by stefanv** [comment]
Rather bizarre!

I don't notice such a large difference in times, but I also operated on a random image, which isn't very typical:

```
In [24]: %timeit ski.morphology.closing(bimage)
2.27 ms ± 24.4 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [25]: %timeit ski.morphology.binary_closing(bimage)
4.63 ms ± 30.7 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

The results are identical. The result is also reproducible on larger images:

```
In [7]: %timeit ski.morphology.closing(bimage)
38.2 ms ± 783 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [8]: %timeit ski.morphology.binary_closing(bimage)
76 ms ± 2.75 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

**Comment by stefanv** [comment]
I wonder if this is a regression in SciPy; it seems unlikely we would have written the guidance if it weren't true at the time. SciPy's docs state the same!

In terms of what to do, easiest for now may be to either (a) deprecate `binary_dilation/closing` or to simply have those functions dispatch to `dilation/closing` and remove the offending documentation.

**Comment by vali19th** [comment]
I think the large kernel makes the timing differences more extreme

**Comment by lagru** [comment]
Agreed, dispatching in binary_dilation/closing to the faster functions seems like the way to go for now. As well as an update to the docstrings.

**Comment by vali19th** [comment]
Maybe the functions can be marked as deprecated and changed to do the dispatching at the same time. This way, the library users will get the speedup without changing their code.

**Comment by michaelbratsch** [comment]
I tried the dispatching approach and then stumbled upon the same issues as being noted in https://github.com/scikit-image/scikit-image/issues/7238, `dilation` and `binary_dilation` have some inconsistencies in the way they operate, see
> Padding of even-sized SEs

and
 
> Mirroring of SE in dilation

That means that dispatching `binary_dilation` to `dilation` is not possible in a straight forward way. I would expect the same to be true for `binary_erosion`. Spending much time to make it work does not seem appropriate to me because it is code we anyway want to get rid of.

I could provide a pull request to only mark the binary functions deprecated. Any thoughts on that?

**Comment by stefanv** [comment]
Thanks @michaelbratsch, that sounds good to me.

**Comment by Himanshunitrr** [comment]
Could not agree more, was using this for filling holes in segmented masks and it took days for all the patches for a WSI, it will be helpful if a faster implementation is implemented. The "workaround" can be to use cv2.dilate (those are two different operations)

**Comment by lagru** [comment]
:+1: on the deprecation approach. As long as we explain how to use alternatives properly with regards to mirroring and even-sized footprints. :) Feel welcome to request a review from me @michaelbratsch if you make a PR.

**Comment by Schefflera-Arboricola** [comment]
https://github.com/scikit-image/scikit-image/pull/7665
